### PR TITLE
feat: Allow generating html id tag for each accordion item

### PR DESCRIPTION
**Pull request by transferwise-yizhang**
☝️ make the title meaningful and follow conventional commits standards

## ❓ Context  
So that each accordion item can have an id. This would allow deep linking into that accordion's section from the url.

## ✅ Checklist

- [ ] All changes are covered by tests
- [ ] All changes have been crossbrowser checked, especially IE11
- [ ] The changes are covered in docs

### CODE REVIEW

File: packages/components/src/accordion/Accordion.js [hunk 1]
```diff
@@ -38,6 +39,7 @@ class Accordion extends PureComponent {
   render() {
     return this.props.items.map((item, index) => (
       <AccordionItem
+        id={this.props.idFunc && this.props.idFunc(item)}
```
Comment (sergonius) [inline packages/components/src/accordion/Accordion.js:42]: You can use optional chaining here

**Comment by paul-frybarger** [comment]
Hm, looks like optional chaining isn't supported:
`SyntaxError: /home/circleci/neptune-web/packages/components/src/accordion/Accordion.js: Support for the experimental syntax 'optionalChaining' isn't currently enabled (42:23)`

**Comment by paul-frybarger** [comment]
How about this:
- add `id` to the accordion item
- if it exists, use that as the id as well as the react key
  - added benefit of keeping the item order if rearranged

**Comment by sergonius** [comment]
> Hm, looks like optional chaining isn't supported:
> `SyntaxError: /home/circleci/neptune-web/packages/components/src/accordion/Accordion.js: Support for the experimental syntax 'optionalChaining' isn't currently enabled (42:23)`

Yeah, `@babel/preset-env` in this repo needs to be upgraded for that :/

**Pull request merged**